js/okex.js

```diff
@@ -2662,10 +2662,10 @@ module.exports = class okex extends Exchange {
 
     parseMyTrade (pair, market = undefined) {
         // check that trading symbols match in both entries
-        const first = pair[0];
-        const second = pair[1];
-        const firstMarketId = this.safeString (first, 'instrument_id');
-        const secondMarketId = this.safeString (second, 'instrument_id');
+        const userTrade = this.safeValue (pair, 1);
+        const otherTrade = this.safeValue (pair, 0);
```

Comment:
@ndubel i'm not sure about these two lines – namely, the user's trade could be either the first in the pair or the second in the pair, and we have to determine which one is "ours".

Reply:
BTC/USDT pair, buy order: 
```
[
    {
      created_at: '2020-10-23T16:17:29.000Z',
      currency: 'USDT',
      exec_type: 'M',
      fee: '0.00480013',
      instrument_id: 'BTC-USDT',
      ledger_id: '239905165493751868',
      liquidity: 'M',
      order_id: '5815258462837761',
      price: '12777.1',
      product_id: 'BTC-USDT',
      side: 'sell',
      size: '96.00265741',
      timestamp: '2020-10-23T16:17:29.000Z',
      trade_id: '121073008'
    },
    {
      created_at: '2020-10-23T16:17:29.000Z',
      currency: 'BTC',
      exec_type: 'M',
      fee: '0',
      instrument_id: 'BTC-USDT',
      ledger_id: '239905165493751866',
      liquidity: 'M',
      order_id: '5815258462837761',
      price: '12777.1',
      product_id: 'BTC-USDT',
      side: 'buy',
      size: '0.00751365',
      timestamp: '2020-10-23T16:17:29.000Z',
      trade_id: '121073008'
    }
  ]
```
Sell order:
```
[
    {
      created_at: '2020-10-24T14:26:33.000Z',
      currency: 'BTC',
      exec_type: 'M',
      fee: '0.00000037',
      instrument_id: 'BTC-USDT',
      ledger_id: '240239638458486785',
      liquidity: 'M',
      order_id: '5820487544232960',
      price: '12941.7',
      product_id: 'BTC-USDT',
      side: 'sell',
      size: '0.00747659',
      timestamp: '2020-10-24T14:26:33.000Z',
      trade_id: '121156528'
    },
    {
      created_at: '2020-10-24T14:26:33.000Z',
      currency: 'USDT',
      exec_type: 'M',
      fee: '0',
      instrument_id: 'BTC-USDT',
      ledger_id: '240239638458486784',
      liquidity: 'M',
      order_id: '5820487544232960',
      price: '12941.7',
      product_id: 'BTC-USDT',
      side: 'buy',
      size: '96.7597848',
      timestamp: '2020-10-24T14:26:33.000Z',
      trade_id: '121156528'
    }
  ]
```

In my understanding, both of this trades are user's, but in case of `buy` as we mean in ccxt, `quote` currency will be in `sell` part of Okex trade and `base` currency will be in `buy` part (we buy BTC for USDT == we sell USDT for BTC). In case of `sell`, `quote` will be in `buy` and `base` will be in `sell` (we sell BTC for USDT == we buy USDT for BTC).
So the `otherTrade` it's also user trade.
`buy` order always has `ledger_id` smaller than `sell` order, so in this case it will always be with index `1` in this array. 
Most of properties are identical in both trades, but we need to find out only `side`, `amount`, `cost` and `fee`.
In case of `buy` (in ccxt meaning) we will see  `side: 'buy'` and `currency: 'BTC'` in part of order with smaller `ledger_id` (now we call it `userTrade`). So we need to take `amount` from this part and cost from other part (aka `otherTrade`).
In case of `sell` vice versa.

We do it here:
```
const receivedCurrencyId = this.safeString (userTrade, 'currency');
if (receivedCurrencyId === quoteId) {
     side = 'sell';
     amount = this.safeFloat (otherTrade, 'size');
     cost = this.safeFloat (userTrade, 'size');
} else {
     side = 'buy';
     amount = this.safeFloat (userTrade, 'size');
     cost = this.safeFloat (otherTrade, 'size');
}
```

Sorry if I am not very clear explaining.

Reply:
Maybe it can be done in some other more elegant way but the main idea is that it's no need to use `fee` to determine was the trade `sell` or `buy`

Reply:
@ndubel thanks for the clarification, everything makes sense, however, we determine the receivedCurrency from the userTrade, which may not always be the `pair[1]`, could be `pair[0]` as well, no? Is there a guarantee anywhere?

Reply:
I see several examples on different pair in all cases `userTrade` was always `pair[1]` (with smaller `ledger_id`, `buy` part of order), but maybe the most reliable way is to choose for `userTrade` the trade with `buy` `side`. Also maybe it should be renamed to `buyTrade` or something like this, because both trades are user in fact

Reply:
@ndubel

I'd rather restore the previous logic that determined the userTrade from quote currency.
If the user sells, the received currency is the base currency.
If the user buys, the received currency is the quote currency.

If an entry in a pair has side = 'buy' and currency = quote – that entry is lying about the user's side (the info in the side field is false for the user requesting the ledger).
If an entry in a pair has side = 'sell' and currency = quote – that entry is telling the truth about the user's side (the info in the side field is true for the user requesting the ledger).
If an entry in a pair has side = 'sell' and currency = base – that entry is lying about the user's side (the info in the side field is false for the user requesting the ledger).
If an entry in a pair has side = 'buy' and currency = base – that entry is telling the truth about the user's side (the info in the side field is true for the user requesting the ledger).

Based on that we can determine the actual side of the user, right? And a corresponding trade should indicate the fee, no?

Reply:
@ndubel I can see that the above does not hold for the buy-order example you've posted above... We can restore the old logic that determines the side, but still keep the new logic that runs through all fees in a pair.

Reply:
> If an entry in a pair has side = 'buy' and currency = quote – that entry is lying about the user's side (the info in the side field is false for the user requesting the ledger).
> If an entry in a pair has side = 'sell' and currency = quote – that entry is telling the truth about the user's side (the info in the side field is true for the user requesting the ledger).
> If an entry in a pair has side = 'sell' and currency = base – that entry is lying about the user's side (the info in the side field is false for the user requesting the ledger).
> If an entry in a pair has side = 'buy' and currency = base – that entry is telling the truth about the user's side (the info in the side field is true for the user requesting the ledger).

In 2 and 3 no, if `sell` and `quote` the entry is lying, if `sell` and `base` the entry is telling truth, if I understood you correctly.
Shortly:
`buy`, `quote` = `sell`
`sell`, `quote` = `buy`
`sell`, `base` = `sell`
`buy`, `base` = `buy`

In examples above i give trades with rebate.
I also put here trades with fee deduction and trades with zero fee:

BTC/USDT, buy with fee deduction:
```
{
      created_at: '2020-10-14T14:26:49.000Z',
      currency: 'USDT',
      exec_type: 'M',
      fee: '0',
      instrument_id: 'BTC-USDT',
      ledger_id: '236615827360141317',
      liquidity: 'M',
      order_id: '5763865474391050',
      price: '11374.1',
      product_id: 'BTC-USDT',
      side: 'sell',
      size: '94.57655142',
      timestamp: '2020-10-14T14:26:49.000Z',
      trade_id: '119653117'
},
{
      created_at: '2020-10-14T14:26:49.000Z',
      currency: 'BTC',
      exec_type: 'M',
      fee: '-0.00000499',
      instrument_id: 'BTC-USDT',
      ledger_id: '236615827360141316',
      liquidity: 'M',
      order_id: '5763865474391050',
      price: '11374.1',
      product_id: 'BTC-USDT',
      side: 'buy',
      size: '0.00831508',
      timestamp: '2020-10-14T14:26:49.000Z',
      trade_id: '119653117'
}
```

Sell with fee deduction:
```
{
      created_at: '2020-10-15T12:25:33.000Z',
      currency: 'BTC',
      exec_type: 'M',
      fee: '0',
      instrument_id: 'BTC-USDT',
      ledger_id: '236947697125904385',
      liquidity: 'M',
      order_id: '5769051009541120',
      price: '11381.6',
      product_id: 'BTC-USDT',
      side: 'sell',
      size: '0.00831011',
      timestamp: '2020-10-15T12:25:33.000Z',
      trade_id: '119799459'
},
{
      created_at: '2020-10-15T12:25:33.000Z',
      currency: 'USDT',
      exec_type: 'M',
      fee: '-0.05674941',
      instrument_id: 'BTC-USDT',
      ledger_id: '236947697125904384',
      liquidity: 'M',
      order_id: '5769051009541120',
      price: '11381.6',
      product_id: 'BTC-USDT',
      side: 'buy',
      size: '94.58234797',
      timestamp: '2020-10-15T12:25:33.000Z',
      trade_id: '119799459'
}
```

UNI/USDT, buy with zero fee:
```
{
      created_at: '2020-10-13T06:38:43.000Z',
      currency: 'USDT',
      exec_type: 'M',
      fee: '0',
      instrument_id: 'UNI-USDT',
      ledger_id: '236135637186072577',
      liquidity: 'M',
      order_id: '5756362609676288',
      price: '3.427',
      product_id: 'UNI-USDT',
      side: 'sell',
      size: '91.51121527',
      timestamp: '2020-10-13T06:38:43.000Z',
      trade_id: '2622421'
},
{
      created_at: '2020-10-13T06:38:43.000Z',
      currency: 'UNI',
      exec_type: 'M',
      fee: '0',
      instrument_id: 'UNI-USDT',
      ledger_id: '236135637186072576',
      liquidity: 'M',
      order_id: '5756362609676288',
      price: '3.427',
      product_id: 'UNI-USDT',
      side: 'buy',
      size: '26.70301',
      timestamp: '2020-10-13T06:38:43.000Z',
      trade_id: '2622421'
}
```

Sell with zero fee:
```
{
      created_at: '2020-10-13T05:09:58.000Z',
      currency: 'UNI',
      exec_type: 'M',
      fee: '0',
      instrument_id: 'UNI-USDT',
      ledger_id: '236113300894076929',
      liquidity: 'M',
      order_id: '5756013617172480',
      price: '3.465',
      product_id: 'UNI-USDT',
      side: 'sell',
      size: '26.41016',
      timestamp: '2020-10-13T05:09:58.000Z',
      trade_id: '2620873'
},
{
      created_at: '2020-10-13T05:09:58.000Z',
      currency: 'USDT',
      exec_type: 'M',
      fee: '0',
      instrument_id: 'UNI-USDT',
      ledger_id: '236113300894076928',
      liquidity: 'M',
      order_id: '5756013617172480',
      price: '3.465',
      product_id: 'UNI-USDT',
      side: 'buy',
      size: '91.5112044',
      timestamp: '2020-10-13T05:09:58.000Z',
      trade_id: '2620873'
}
```

Reply:
@ndubel thx for your clarifications, i'll add a few edits and will merge it shortly.